### PR TITLE
#44 Sampling tasks from user data

### PR DESCRIPTION
**Pull request by mjagkow**
Tasks are now sampled randomly from user-uploaded data.

- Queries to fetch random post threads and conversations
- Moved "frontend task message" id to WorkPackage
- Added references for thread_id, parent_post_id to WorkPackage
- Added "ack" and "done" flags to WorkPackage
- PostReaction FK changed to WorkPackage
- Added depth and children_count to Post

### CODE REVIEW

File: backend/oasst_backend/models/post.py [hunk 1]
```diff
@@ -31,3 +31,5 @@ class Post(SQLModel, table=True):
     )
     payload_type: str = Field(nullable=False, max_length=200)
     payload: PayloadContainer = Field(sa_column=sa.Column(payload_column_type(PayloadContainer), nullable=True))
+    depth: int = Field(sa_column=sa.Column(sa.Integer, default=0, server_default=sa.text("0"), nullable=False))
```
Comment (yk) [inline backend/oasst_backend/models/post.py:34]: do we really need all these attributes?
it seems like sqlmodel might do the right thing with `depth: int = 0` without any field definition at all, no?

Comment (mjagkow) [reply backend/oasst_backend/models/post.py:34]: There was an issue adding default boolean value (ack, done), the value did not propagate properly to existing rows. Solution was to specify server_default, so I did it the same way to other columns.

**Comment by yk** [comment]
LGTM, but I'll leave the review to @andreaskoepf bc more familiar with this code

**Review by andreaskoepf — APPROVED** [review]
My thoughts:
1. depth and children_count are incrementally calculated, e.g. we must from now on not directly delete elements from post without going through python code that updates the parent element.
2. regarding sampling & `order by random()` I have read different reports about its performance online (some say it "performs a full table scan and also ordering. Therefore this method is not preferred for tables with large number of rows because of performance reasons." ..) Postgres also supports TABLESAMPLE, which is probably faster.

In general we will need very soon a more sophisticated sampling function that takes a weights into account.

**Comment by andreaskoepf** [comment]
linking #44

**Pull request merged**